frontend/src/stores/user.js

```diff
@@ -0,0 +1,25 @@
+import { computed, ref } from 'vue';
+import { defineStore } from 'pinia';
+import userService from '@/api/users';
+import { useStorage } from '@vueuse/core';
+
+export const useUserStore = defineStore('user', () => {
+  const user = ref(useStorage('user', {}));
+  const token = ref(useStorage('token', ''));
+
+  const isLoggedIn = computed(() => {
+    return !!user.value.id;
+  });
+
+  async function loginUser(jwtToken) {
+    user.value = await userService.getUserProfileFromToken(jwtToken);
+    token.value = jwtToken;
```

Comment:
If this line were above the previous one, could we then just call `getUserProfile`?

Reply:
I tried that but it didn't work, probably because useStorage does not save value immediately

Reply:
Interesting. I'll take a look at their documentation and code.

Reply:
I took a quick look at their documentation, code, and Github issues. It seems some people have reported similar issues. But thinking about how to approach this, I realized that this info is already available in the store and the local storage should only play the part on the initial page load. To restore the previous state. The store should always have the latest. So something like the code below works and should cover the use case.

```js
import { useUserStore } from '@/stores/user';

function updateOptions(options) {
  const store = useUserStore();
  const update = { ...options };

  if (store.token) {
    update.headers = {
      ...update.headers,
      Authorization: `Bearer ${store.token}`,
    };
  }
  return update;
}
```
Note: it is based on the previous version of the code.

Reply:
fixed!